lib/executable-config/inject/src/Obelisk/ExecutableConfig/Inject.hs

```diff
@@ -9,9 +17,28 @@ import qualified Data.Text.Encoding as T
 import Reflex.Dom
 import System.FilePath ((</>))
 
-inject :: FilePath -> IO ByteString
-inject item = do
-  f <- BS.readFile $ "config" </> item
+import Obelisk.ExecutableConfig.Types
+
+data InjectionDisallowed = forall config. ObeliskConfig config
+  => InjectionDisallowed (ConfigLocation config)
+
+deriving instance Show InjectionDisallowed
+instance Exception InjectionDisallowed
+
+inject :: forall config. ObeliskConfig config => config -> IO ByteString
+inject cfg = do
+  let loc = configLocation @config
+  unless (canInject loc) $
+    throw $ InjectionDisallowed loc
   fmap snd $ renderStatic $
-    elAttr "script" ("type" =: "text/plain" <> "id" =: ("config-" <> T.pack item)) $
-      text $ T.decodeUtf8 f
+    elAttr "script" ("type" =: "text/plain" <> "id" =: key) $
+      text $ encodeConfig cfg
+  where
+    key = "injected-" <> T.pack (getConfigPath @config)
+
+-- | Check if we are allowed to inject the given config into the DOM.
+-- TODO: Can we perform this check at the compile time?
+canInject :: ObeliskConfig config => ConfigLocation config -> Bool
+canInject = \case
+  ConfigLocation CabalProject_Backend _ -> False
+  _ -> True
```

Comment:
If the user tries to inject a backend config, this would throw an error at runtime. I was wondering if we can make that a compile error (so it gets caught early). @ryantrinkle Is this worth doing? It would, as one solution, involve a new type class `ObeliskInjectableConfig` (derivative of `ObeliskConfig`).